src/workerd/api/sockets.c++

```diff
@@ -9,8 +9,47 @@
 namespace workerd::api {
 
 
+bool isValidAddress(kj::StringPtr address) {
+  // This function performs some basic length and characters checks, it does not guarantee that
+  // the specified address is a valid domain. It should only be used to reject malicious
+  // addresses.
+  if (address.size() > (255 + 6)) {
+    // RFC1035 states that maximum domain name length is 255 octets. But we also add 6 to support
+    // port numbers in the address.
+    //
+    // IP addresses are always shorter, so we take the max domain length instead.
+    return false;
+  }
+
+  for (int i = 0; i < address.size(); i++) {
+    switch (address[i]) {
+      case '-':
+      case '.':
+      case ':':
+      case '_':
+      case '[': case ']': // For IPv6.
```

Comment:
@jasnell Is there any consensus on how IPv6 addresses should be represented? This code will end up relying on KJ's syntax which is like `[1234:5678::abcd]:12345` -- i.e. the address goes in brackets, so that the `:`'s in the address aren't misinterpreted as being the port delimiter. But I'm not really sure if that syntax is standard enough to be appropriate...

Reply:
Ah, looks like [RFC 2732](https://www.rfc-editor.org/rfc/rfc2732) says this is actually a standard, not something I made up. OK that seems good then.

(It was 10 years ago, I can't remember.)

Reply:
Yep, follow 2732 on this.